From efb3523a9851e2feb1e98f5a5862f1d858930ef8 Mon Sep 17 00:00:00 2001
From: Shane McDonald <mcdonald.shane@gmail.com>
Date: Mon, 27 Apr 2009 23:50:21 -0600
Subject: [PATCH 044/117] MIPS: MSP71xx: Resolve multiple definition of plat_timer_setup

commit efb3523a9851e2feb1e98f5a5862f1d858930ef8 from linux-mips

There have been a number of compile problems with the msp71xx configuration ever since
it was included in the linux-mips.org repository.  This patch resolves the "multiple
definition of plat_timer_setup" problem, and creates the required get_c0_compare_int
function.

This patch has been compile-tested against the current HEAD.

Signed-off-by: Shane McDonald <mcdonald.shane@gmail.com>
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit a6d0acc0e87c245b50784142ceff61cdb061342c)
---
 arch/mips/pmc-sierra/msp71xx/msp_time.c |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/arch/mips/pmc-sierra/msp71xx/msp_time.c b/arch/mips/pmc-sierra/msp71xx/msp_time.c
index 7cfeda5..cca64e1 100644
--- a/arch/mips/pmc-sierra/msp71xx/msp_time.c
+++ b/arch/mips/pmc-sierra/msp71xx/msp_time.c
@@ -81,10 +81,7 @@ void __init plat_time_init(void)
 	mips_hpt_frequency = cpu_rate/2;
 }
 
-void __init plat_timer_setup(struct irqaction *irq)
+unsigned int __init get_c0_compare_int(void)
 {
-#ifdef CONFIG_IRQ_MSP_CIC
-	/* we are using the vpe0 counter for timer interrupts */
-	setup_irq(MSP_INT_VPE0_TIMER, irq);
-#endif
+	return MSP_INT_VPE0_TIMER;
 }
-- 
1.7.0

